From 53f9bab6b1cec73486a2f6f28cc6c0f019789b2e Mon Sep 17 00:00:00 2001
From: Chris Diamand <chris@diamand.org>
Date: Thu, 26 Feb 2015 20:28:07 +0000
Subject: [PATCH 08/11] HACK: Don't try and parse the VDSO.

Bad things happen when we encounter linux-vdso.so.1 in
__liballocs_iterate_types(), so skip it.
---
 src/liballocs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/liballocs.c b/src/liballocs.c
index 1403ca8..0922760 100644
--- a/src/liballocs.c
+++ b/src/liballocs.c
@@ -144,6 +144,10 @@ int __liballocs_iterate_types(void *typelib_handle, int (*cb)(struct uniqtype *t
 	struct link_map *h = typelib_handle;
 	unsigned char *load_addr = (unsigned char *) h->l_addr;
 	
+	// HACK: Ignore the VDSO.
+	if (!strcmp(h->l_name, "linux-vdso.so.1"))
+	    return 0;
+
 	/* If load address is greater than STACK_BEGIN, it's suspicious -- 
 	 * perhaps a vdso-like thing. Skip it. The vdso itself is detected
 	 * below (it lives in user memory, but points into kernel memory). */
-- 
2.1.4

